crates/services/p2p/src/p2p_service.rs

```diff
@@ -522,14 +528,24 @@ impl<Codec: NetworkCodec> FuelP2PService<Codec> {
                         ) {
                             let _ = self.swarm.disconnect_peer_id(peer_id);
                         } else if initial_connection {
-                            return Some(FuelP2PEvent::PeerConnected(peer_id))
+                            return Some(FuelP2PEvent::PeerConnected {
+                                peer_id,
+                                reserved_peers_connected_count: self
+                                    .peer_manager
+                                    .connected_reserved_peers(&peer_id),
+                            })
                         }
                     }
                     PeerReportEvent::PeerDisconnected { peer_id } => {
                         if self.peer_manager.handle_peer_disconnect(peer_id) {
                             let _ = self.swarm.dial(peer_id);
                         }
-                        return Some(FuelP2PEvent::PeerDisconnected(peer_id))
+                        return Some(FuelP2PEvent::PeerDisconnected {
```

Comment:
I think it would be better if the `PeerManager` itself provides a subscription. In this case, you don't need `.connected_reserved_peers(&peer_id)` function and the event will be simple as possible without propagation of the new information.